src/Algebra/Properties/Semiring/GCD.agda

```diff
@@ -7,24 +7,45 @@
 {-# OPTIONS --without-K --safe #-}
 
 open import Algebra using (Semiring)
-import Algebra.Properties.Monoid.Divisibility as MonoidDiv
-open import Data.Product using (_,_)
+import Algebra.Primality as Primality
 open import Data.Sum.Base using (_⊎_; inj₁; inj₂)
 
 module Algebra.Properties.Semiring.GCD {a ℓ} (R : Semiring a ℓ) where
 
 open Semiring R
+open Primality _≈_ _*_ 0# 1# using (Coprime)
+-- open import Algebra.Properties.Monoid.Divisibility *-monoid using (∣-refl)
+
 open import Algebra.Properties.Semiring.Divisibility R
 
 ------------------------------------------------------------------------
 -- Re-exporting definition of GCD
 
 open import Algebra.Divisibility _≈_ _*_ public
-  using (IsGCD; gcdᶜ)
+  using (IsGCD; isGCDᶜ)
 
 ------------------------------------------------------------------------
 -- Properties of GCD
 
+isGCD-0-x-x : ∀ x → IsGCD 0# x x
+isGCD-0-x-x x =  isGCDᶜ (_∣0 x) ∣-refl (λ _ y∣x → y∣x)
+
+isGCD-x-0-x : ∀ x → IsGCD x 0# x
+isGCD-x-0-x x =  isGCDᶜ ∣-refl (_∣0 x) (λ y∣x _ → y∣x)
+
 x≉0⊎y≉0⇒gcd≉0 : ∀ {x y d} → IsGCD x y d → x ≉ 0# ⊎ y ≉ 0# → d ≉ 0#
-x≉0⊎y≉0⇒gcd≉0 (gcdᶜ d∣x _ _) (inj₁ x≉0) = x∣y∧y≉0⇒x≉0 d∣x x≉0
-x≉0⊎y≉0⇒gcd≉0 (gcdᶜ _ d∣y _) (inj₂ y≉0) = x∣y∧y≉0⇒x≉0 d∣y y≉0
+x≉0⊎y≉0⇒gcd≉0 (isGCDᶜ d∣x _ _) (inj₁ x≉0) = x∣y∧y≉0⇒x≉0 d∣x x≉0
+x≉0⊎y≉0⇒gcd≉0 (isGCDᶜ _ d∣y _) (inj₂ y≉0) = x∣y∧y≉0⇒x≉0 d∣y y≉0
+
+x≈0∧y≈0⇒gcd≈0 : ∀ {x y d} → IsGCD x y d → x ≈ 0# → y ≈ 0# → d ≈ 0#
+x≈0∧y≈0⇒gcd≈0 (isGCDᶜ _ _ greatest) x≈0 y≈0 =  0∣x⇒x≈0 0∣d
+  where
+  0∣x = ∣-respʳ (sym x≈0) (0# ∣0)
+  0∣y = ∣-respʳ (sym y≈0) (0# ∣0)
+  0∣d = greatest {0#} 0∣x 0∣y
+
+coprime⇒gcd∣1 : ∀ {x y d} → IsGCD x y d → Coprime x y →  d ∣ 1#
```

Comment:
Given the name, the argument order should really be swapped.

Reply:
Done.